### PR TITLE
support build individual module

### LINKED ISSUE #5488: Support build individual module
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [x] Bug
- [ ] Requirement
- [ ] Feature or performance improvement

___
### Question
- What do you want to know?

___
### Bug
- Which version of SkyWalking, OS and JRE?
 current main branch  can't build individual module, 
exception is 
```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  12.094 s
[INFO] Finished at: 2020-09-15T15:34:10+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check (validate) on project apm-agent: Failed during checkstyle configuration: cannot initialize module TreeWalker - cannot initialize module ImportControl - illegal value 'apm-checkstyle/importControl.xml' for property 'file': com.puppycrawl.tools.checkstyle.api.CheckstyleException: Unable to find: apm-checkstyle/importControl.xml -> [Help 1]
[ERROR] 

```
- Which company or project?

- What happened?
If possible, provide a way to reproduce the error. e.g. demo application, component version.
 It is caused by checkstyle file is relative to project root, should be absolute  
___
### Requirement or improvement
- Please describe your requirements or improvement suggestions.

### PR DESCRIPTION
**Pull request by xbkaishui**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bug fix

- Related issues
resolved #5488 
___
### Bug fix
- Bug description.
- How to fix?
In submodule build, using the absolute path to find the check-style file
___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

**Comment by wu-sheng** [comment]
This commend doesn't work locally.

**Comment by xbkaishui** [comment]
I this in my Mac, it works ok,  can you show the details?

**Comment by wu-sheng** [comment]
CI fails too. Even I don't think CI tested this feature. 

> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check (validate) on project apm-ehcache-2.x-plugin: Failed during checkstyle execution: Failed during checkstyle configuration: unable to parse configuration stream: Property ${import.control} has not been set -> [Help 1]

**Property ${import.control} has not been set**, I think this is the reason.

**Comment by kezhenxu94** [comment]
I don't think that's what we recommend, we recommend to use `mvnw` with `-f`, which maybe also **INCONVENIENT** as what you said when the submodule is too **DEEP**, you have `../../../mvnw`, if the pull request doesn't break anything (tests passed), I'm also OK to merge this

**Comment by wu-sheng** [comment]
> if the pull request doesn't break anything (tests passed), I'm also OK to merge this

I am wondering why I can't pass compiling locally. Have a concern about this change

**Comment by xbkaishui** [comment]
let me double check,  
In my local mac, I can switch to any module and just run `mvn clean package` , it will success

屏幕快照 2020-09-15 下午5 32 29

**Comment by wu-sheng** [comment]
`mvn clean package` is sensitive about the maven version, I used that too, but failed.

```
Wu-Sheng-s-MacBook-Pro:ehcache-2.x-plugin wusheng$ mvn package
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building ehcache-2.x-plugin 8.2.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
Downloading: https://repository.apache.org/snapshots/org/apache/skywalking/apm-sniffer/8.2.0-SNAPSHOT/maven-metadata.xml
Downloading: https://repository.apache.org/snapshots/org/apache/skywalking/apm/8.2.0-SNAPSHOT/maven-metadata.xml
Downloading: https://repository.apache.org/snapshots/org/apache/skywalking/apm-commons/8.2.0-SNAPSHOT/maven-metadata.xml
Downloading: https://repository.apache.org/snapshots/org/apache/skywalking/apm-test-tools/8.2.0-SNAPSHOT/maven-metadata.xml
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-maven-version) @ apm-ehcache-2.x-plugin ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-java) @ apm-ehcache-2.x-plugin ---
[INFO] 
[INFO] --- os-maven-plugin:1.6.2:detect (default) @ apm-ehcache-2.x-plugin ---
[INFO] ------------------------------------------------------------------------
[INFO] Detecting the operating system and CPU architecture
[INFO] ------------------------------------------------------------------------
[INFO] os.detected.name: osx
[INFO] os.detected.arch: x86_64
[INFO] os.detected.version: 10.15
[INFO] os.detected.version.major: 10
[INFO] os.detected.version.minor: 15
[INFO] os.detected.classifier: osx-x86_64
[INFO] 
[INFO] --- maven-checkstyle-plugin:3.1.0:check (validate) @ apm-ehcache-2.x-plugin ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 7.388 s
[INFO] Finished at: 2020-09-15T16:24:45+08:00
[INFO] Final Memory: 18M/173M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check (validate) on project apm-ehcache-2.x-plugin: Failed during checkstyle execution: Failed during checkstyle configuration: unable to parse configuration stream: Property ${import.control} has not been set -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Wu-Sheng-s-MacBook-Pro:ehcache-2.x-plugin wusheng$ mvn package
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building ehcache-2.x-plugin 8.2.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-maven-version) @ apm-ehcache-2.x-plugin ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-java) @ apm-ehcache-2.x-plugin ---
[INFO] 
[INFO] --- os-maven-plugin:1.6.2:detect (default) @ apm-ehcache-2.x-plugin ---
[INFO] ------------------------------------------------------------------------
[INFO] Detecting the operating system and CPU architecture
[INFO] ------------------------------------------------------------------------
[INFO] os.detected.name: osx
[INFO] os.detected.arch: x86_64
[INFO] os.detected.version: 10.15
[INFO] os.detected.version.major: 10
[INFO] os.detected.version.minor: 15
[INFO] os.detected.classifier: osx-x86_64
[INFO] 
[INFO] --- maven-checkstyle-plugin:3.1.0:check (validate) @ apm-ehcache-2.x-plugin ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.672 s
[INFO] Finished at: 2020-09-15T16:25:54+08:00
[INFO] Final Memory: 15M/253M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check (validate) on project apm-ehcache-2.x-plugin: Failed during checkstyle execution: Failed during checkstyle configuration: unable to parse configuration stream: Property ${import.control} has not been set -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Wu-Sheng-s-MacBook-Pro:ehcache-2.x-plugin wusheng$ mvn -version
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-11T00:41:47+08:00)
Maven home: /Users/wusheng/Documents/applications/apache-maven-3.3.9
Java version: 1.8.0_91, vendor: Oracle Corporation
Java home: /Users/wusheng/Documents/applications/jdk1.8.0_91.jdk/Contents/Home/jre
Default locale: zh_CN, platform encoding: UTF-8
OS name: "mac os x", version: "10.15.6", arch: "x86_64", family: "mac"
```

That is also why, @kezhenxu94 said, you will face another deep path because `mvnw`.

**Comment by xbkaishui** [comment]
The maven version should greater than **3.6**,  I also saw the e2e/pom.xml use this property 
https://github.com/apache/skywalking/blob/224af7f6256c7926455b5ecc5acd2cf9047e81ea/test/e2e/pom.xml#L155

**Comment by wu-sheng** [comment]
mvnw provides that. But, local maven may be different version on different laptop.

**Comment by xbkaishui** [comment]
Can we enforce the maven version in main pom.xml , just like enforce the  jdk version must greater than 1.8

**Comment by wu-sheng** [comment]
Is that possible?

**Comment by xbkaishui** [comment]
Yes, I can change it

**Comment by xbkaishui** [comment]
Hi @wu-sheng , added rule to check maven must greater than 3.6, please check in your local env

File: pom.xml [hunk 1]
```diff
@@ -383,12 +383,16 @@
...
                         </goals>
+                        <phase>validate</phase>
                         <configuration>
                             <rules>
                                 <requireJavaVersion>
                                     <!-- Build has not yet been updated for Java 9+ -->
                                     <version>1.8</version>
                                 </requireJavaVersion>
+                                <requireMavenVersion>
+                                    <version>3.6</version>
```
Comment (wu-sheng) [inline pom.xml:394]: This makes you need to update the compiling doc

https://github.com/apache/skywalking/blob/master/docs/en/guides/How-to-build.md#build-from-github

Comment (wu-sheng) [reply pom.xml:394]: At least 3 places should be updated.
1. Prepare git, JDK8, and Maven 3.6+
1. Add document about compiling a single plugin in the `Advanced compile` part.
1. Remove `Setup your Eclipse IDE` section, it is out of the update.

**Review by wu-sheng — COMMENTED** [review]
It works, the document should be updated.

File: pom.xml (hunk 1 above)
Comment (xbkaishui) [reply pom.xml:394]: Done, Please help to check (my poor english)

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**